File: backends/fastapi/tests/fixtures/agent_events.py
"""Deterministic AgentEvent fixture data for agent event router tests.

``create_task_events(repo)`` populates an in-memory AgentRepository with a
canonical set of events covering every supported event_type so that router
and service tests can rely on predictable sequence numbers and event shapes
without running the real agent loop.

Fixed identifiers
-----------------
SESSION_ID = "test-session"
TASK_ID    = "test-task"

Event sequence
--------------
1  agent_started      — loop configuration
2  assistant_output   — LLM text fragment
3  tool_call          — read_file invocation
4  tool_result        — file content returned
5  approval_requested — file.write requires human approval
6  error              — terminal failure event (provider timeout)
7  completed          — terminal success event (overrides error in separate fixture)
"""

from __future__ import annotations

from app.agent.repository import AgentRepository

SESSION_ID = "test-session"
TASK_ID = "test-task"

# Deterministic event IDs so assertions can reference them by name if needed
_EVT_IDS = {
    "agent_started": "evt-001",
    "assistant_output": "evt-002",
    "tool_call": "evt-003",
    "tool_result": "evt-004",
    "approval_requested": "evt-005",
    "error": "evt-006",
}

_COMPLETED_ID = "evt-007"

_EVENTS = [
    (
        "evt-001",
        "agent_started",
        {"max_iterations": 20, "max_tool_calls": 50, "task_timeout_seconds": 300.0},
    ),
    (
        "evt-002",
        "assistant_output",
        {"content": "I will start by reading the login handler."},
    ),
    (
        "evt-003",
        "tool_call",
        {"tool_name": "repo.read", "tool_input": {"path": "src/auth/login.py"}, "sequence": 1},
    ),
    (
        "evt-004",
        "tool_result",
        {"tool_name": "repo.read", "output": {"content": "def login(request): ..."}, "sequence": 1},
    ),
    (
        "evt-005",
        "approval_requested",
        {
            "approval_id": "appr-001",
            "tool_name": "file.write",
            "risk": "mutating",
            "fingerprint": "sha256:abc123",
        },
    ),
    (
        "evt-006",
        "error",
        {"error_type": "provider_unavailable", "message": "Provider timeout after 30s"},
    ),
]

_COMPLETED_EVENTS = _EVENTS[:-1] + [
    (
        _COMPLETED_ID,
        "completed",
        {"content": "Refactoring applied successfully.", "iterations": 3, "tool_calls": 1},
    ),
]


def create_task_events(
    repo: AgentRepository,
    *,
    terminal: str = "error",
) -> tuple[str, str]:
    """Populate *repo* with a canonical set of task events.

    Parameters
    ----------
    repo:
        An ``AgentRepository`` instance (typically in-memory) to populate.
    terminal:
        ``"error"`` (default) ends the sequence with an error event;
        ``"completed"`` ends with a completed event.

    Returns
    -------
    (session_id, task_id) — the fixed identifiers used for all events.
    """
    repo.create_session(SESSION_ID)
    repo.create_task(TASK_ID, SESSION_ID, "Refactor the login handler")

    events = _COMPLETED_EVENTS if terminal == "completed" else _EVENTS
    for event_id, event_type, payload in events:
        repo.append_event(
            event_id=event_id,
            session_id=SESSION_ID,
            task_id=TASK_ID,
            event_type=event_type,
            payload=payload,
        )
    return SESSION_ID, TASK_ID

File: apps/mobile/test/fixtures/agent-events.ts
/**
 * Deterministic AgentEvent replay fixtures for mobile tests.
 *
 * Covers sequences 1–6 with all required event types:
 *   agent_started, assistant_output, tool_call, tool_result,
 *   approval_requested, error (and alternate completed set).
 */

import type { AgentEventResponse } from "../../lib/api";

export const SESSION_ID = "test-session";
export const TASK_ID = "test-task";
export const APPROVAL_ID = "appr-001";

export const agentStartedEvent: AgentEventResponse = {
  event_id: "evt-001",
  session_id: SESSION_ID,
  task_id: TASK_ID,
  sequence: 1,
  event_type: "agent_started",
  payload: { max_iterations: 20, max_tool_calls: 50, task_timeout_seconds: 300 },
  created_at: "2026-09-26T10:00:01Z",
};

export const assistantOutputEvent: AgentEventResponse = {
  event_id: "evt-002",
  session_id: SESSION_ID,
  task_id: TASK_ID,
  sequence: 2,
  event_type: "assistant_output",
  payload: { content: "I will start by reading the auth module." },
  created_at: "2026-09-26T10:00:02Z",
};

export const toolCallEvent: AgentEventResponse = {
  event_id: "evt-003",
  session_id: SESSION_ID,
  task_id: TASK_ID,
  sequence: 3,
  event_type: "tool_call",
  payload: {
    tool_name: "repo.read",
    tool_input: { path: "src/auth/login.py" },
    sequence: 1,
  },
  created_at: "2026-09-26T10:00:03Z",
};

export const toolResultEvent: AgentEventResponse = {
  event_id: "evt-004",
  session_id: SESSION_ID,
  task_id: TASK_ID,
  sequence: 4,
  event_type: "tool_result",
  payload: {
    tool_name: "repo.read",
    output: { content: "def login(request): ..." },
    sequence: 1,
  },
  created_at: "2026-09-26T10:00:04Z",
};

export const approvalRequestedEvent: AgentEventResponse = {
  event_id: "evt-005",
  session_id: SESSION_ID,
  task_id: TASK_ID,
  sequence: 5,
  event_type: "approval_requested",
  payload: {
    approval_id: APPROVAL_ID,
    tool_name: "file.write",
    risk: "mutating",
    fingerprint: "sha256:abc123",
  },
  created_at: "2026-09-26T10:00:05Z",
};

export const errorEvent: AgentEventResponse = {
  event_id: "evt-006",
  session_id: SESSION_ID,
  task_id: TASK_ID,
  sequence: 6,
  event_type: "error",
  payload: {
    error_type: "provider_unavailable",
    message: "Provider timeout after 30s",
  },
  created_at: "2026-09-26T10:00:06Z",
};

export const completedEvent: AgentEventResponse = {
  event_id: "evt-007",
  session_id: SESSION_ID,
  task_id: TASK_ID,
  sequence: 6,
  event_type: "completed",
  payload: {
    content: "Refactoring applied successfully.",
    iterations: 3,
    tool_calls: 1,
  },
  created_at: "2026-09-26T10:00:06Z",
};

/** Ordered sequence 1–6 ending with error. */
export const errorReplay: AgentEventResponse[] = [
  agentStartedEvent,
  assistantOutputEvent,
  toolCallEvent,
  toolResultEvent,
  approvalRequestedEvent,
  errorEvent,
];

/** Ordered sequence 1–6 ending with completed. */
export const completedReplay: AgentEventResponse[] = [
  agentStartedEvent,
  assistantOutputEvent,
  toolCallEvent,
  toolResultEvent,
  approvalRequestedEvent,
  completedEvent,
];
